### PR TITLE
175 add thumbnail view

### PR DESCRIPTION
**Pull request by michael-conway**
bring gallery api controller into mainline code. Maven artifacts are updated and gallery API controller calls a live rule...

Screen Shot 2021-02-01 at 11 18 54 AM

```
irods_policy_list_thumbnails_for_logical_path(*logical_path,*length,*offset,*out)
{
    *out = '{ "location": "testLocation", "items": [ { "id": 1, "name": "sample1.png", "lastModified": "2020-12-14", "thumbnails": "./img/sample1.jpg" }, { "id": 2, "name": "sample2.png", "lastModified": "2020-12-14", "thumbnails": "./img/sample2.jpg" }, { "id": 3, "name": "sample3.png", "lastModified": "2020-12-14", "thumbnails": "./img/sample3.jpg" }] }';
}


```

### CODE REVIEW

**Comment by michael-conway** [comment]
...might not be time yet for a merge but on this side it's done...

**Comment by trel** [comment]
let me rebase and get rid of the merge commit...

File: src/metalnx-web/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "buildx": "npm -v",
-    "build": "vue-cli-service build --mode production",
+    "build": "vue-cli-service build --mode development",
```
Comment (trel) [inline src/metalnx-web/package.json:8]: is this now correct?   should we put it back to production before merging to master?

**Comment by trel** [comment]
also +57 on the codacy

**Comment by michael-conway** [comment]
Thinking about it perhaps we leave in the branch and let Bo do the PR when
he;s done?

**Comment by trel** [comment]
Sure - but we're IN the PR now.   I'll let it breathe for a day or so.

**Comment by connectbo** [comment]
@michael-conway Hi Mike, I pulled down the code and might have several questions... Correct me if I am wrong :)

- In terms of live, if the url param 'path' changes, will the json file galleryApiController send back reflects the changes?
- I tried to add some new files into my path in collectionBrowser, and the 'item' in json files wasn't updating? Not sure if that only happens to me.
- Can we have another json object key under each file for 'hover'? So that we can show some data when mouseover.. See below.. Thanks!
  
  location: "testLocation",
      items: [
        {
          id: 1,
          name: "sample1.png",
          lastModified: "2020-12-14",
          thumbnails: "./img/sample1.png",
          hover: {
            filesize: "10MB",
            filetype: "Portable Network Graphics",
          },
        }

**Comment by michael-conway** [comment]
You are correct. I believe at this point the rule is returning a hard-coded
response no matter what you provide in path. I think Kory is working on a
live impl of the rule. You can ensure this by editing core.re and changing
the data to make sure you've got things lined up and calling all the way
down, but this looks correct. The path, length, and offset are passed to
the rule for paging as you suspect.

**Comment by trel** [comment]
@connectbo is this actually rendering the thumbnail paths sent back from the rule?  or is it still always just rendering a placeholder icon?

**Pull request merged**